checker/src/main/java/org/checkerframework/checker/resourceleak/ResourceLeakVisitor.java

```diff
@@ -467,6 +471,23 @@ private void checkOwningField(Element field) {
     }
   }
 
+  /**
+   * Determine if the given expression <code>e</code> refers to <code>this.field</code>.
+   *
+   * @param e the expression
+   * @param field the field
+   * @return true if <code>e</code> refers to <code>this.field</code>
+   */
+  private boolean expressionEqualsField(String e, VariableElement field) {
+    try {
+      JavaExpression je = StringToJavaExpression.atFieldDecl(e, field, this.checker);
+      return je instanceof FieldAccess && ((FieldAccess) je).getField().equals(field);
```

Comment:
Do we also need to check if the receiver is `this`?

Reply:
`StringToJavaExpression` should handle that for us: it uses the supplied location (`field`, the declaration of the owning field, in this case) and the given string to find the appropriate program element, and associates that with the result.

A potential point of confusion in my implementation is that `field` is used for two purpose:
* it's the element corresponding to the owning field, so the object that `StringToJavaExpression` converts the input string into is compared against it
* it is used as the location at which `StringToJavaExpression` should consider the string `e` to occur in the code: that is, the context

Reply:
Could I write something like

```java
@MustCall("close")
class SneakyDestructor {
  private final @Owning Closeable resource;

  // ...

  @EnsuresCalledMethods(value = "#1.resource", methods = "close")
  void close(SneakyDestructor other) {
    other.close();
  }
}
```

?

In that case, the field referred to by the expression really is `SneakyDestructor.resource`, but the destructor clearly doesn't close its own `@Owning` field.

Reply:
I understand what you mean now, thanks. I fleshed out your test case and added it to this PR, and a bit to my surprise the tests are passing.